openmc_dagmc_wrapper/tallies/mesh_tallies/mesh_tally_3D.py

```diff
@@ -0,0 +1,69 @@
+from typing import List, Tuple
+
+import openmc
+
+# TODO:
+# i don't understand why . here works but not openmc_dagmc_wrapper
+from . import MeshTally
+
+
+# my_tally = odw.Tally("(n,Xt)")
+# my_mesh = openmc.RegularMesh(name="3d_mesh")
+# my_mesh.dimension = (100, 100, 100)
+# my_mesh.lower_left = (0, 0, 0)
+# my_mesh.upper_right = (10, 10, 10)
+# my_tally.filters.append(openmc.MeshFilter(my_mesh))
+
+
+# my_tally = odw.MeshTally3D(
+#     tally_type="(n,Xt)",
+#     bounding_box=[(0, 0, 0), (10, 10, 10)],
+#     mesh_resolution=(100, 100, 100)
+# )
+
+class MeshTally3D(MeshTally):
+    def __init__(
+        self,
+        tally_type: str,
+        bounding_box: List[Tuple[float]],
+        mesh_resolution=(100, 100, 100),
+        **kwargs
+    ):
+        mesh = self.create_mesh(mesh_resolution, bounding_box)
+        super().__init__(tally_type, mesh, **kwargs)
+        self.name = self.tally_type + "_on_3D_mesh"
```

Comment:
Having the name attribute defaulting to this value can even be dangerous:

Say a user wants to have the score `'heating'` on two different `RegularMesh`.
```python
tally_1 = MeshTally3D("heating", mesh_resolution=(100, 100, 100), bounding_box=[point1, point2])
tally_2 = MeshTally3D("heating", mesh_resolution=(10,   10,    10),  bounding_box=[point1, point2])
```
I think it's likely that `tally_1.name == tally_2.name`.
This could have consequences in the post-processing part of the workflow!